### PR TITLE
Annotation border styles

### LINKED ISSUE #5694: Hyperlink displays incorrectly
http://www.karger.com/Article/Pdf/368404

The above PDF displays the hyperlink at the top incorrectly.

Using Chrome 39, Windows 8.1 x64 and PDF.js 1.0.1132 (latest)

PDF.js
image

Adobe Acrobat
image

### PR DESCRIPTION
**Pull request by timvandermeij**
This PR implements annotation border styles according to the official PDF specification at https://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/pdf_reference_1-7.pdf, section 8.4.3.

Most of the code is rewriting the old code, but there are some minor additions in order to adhere better to the specification. These are:
- Support for horizontal and vertical corner radii (just a few lines of code)
- Support for dashed and underline border styles (two lines of code in `annotation_helper.js`); this improves all ArXiv PDFs such as http://www.cs.columbia.edu/~djhsu/papers/ilovetoconbandits-arxiv.pdf (dashed border on the left) and fixes #5694 (underline at the top)

Unimplemented border styles have been indicated using `warn()` so it is clear that these are missing (and there are barely used in PDFs on the web). I have added documentation and unit/regression tests to increase testability of the annotation layer and to avoid regressions.

### CODE REVIEW

**Comment by timvandermeij** [comment]
This patch has only been rebased onto the current master. No changes have been made to the code.

**Comment by CodingFabian** [comment]
thanks @timvandermeij for picking this up. i thought it landed already.

File: src/core/annotation.js [hunk 1]
```diff
@@ -148,6 +112,57 @@ var Annotation = (function AnnotationClosure() {
...
 
   Annotation.prototype = {
+    /**
+     * Set the border style (as AnnotationBorderStyle object).
+     *
+     * @public
+     * @memberof Annotation
+     * @param borderStyle {Dict} The border style dictionary
+     */
+    setBorderStyle: function Annotation_setBorderStyle(borderStyle) {
```
Comment (brendandahl) [inline src/core/annotation.js]: It seems a bit weird to have a get and set that return/take different things. What about: 

1) In the constructor create the AnnotationBorderStyle and set it to borderStyle
2) have setBorderStyle become updateBorderStyle
3) remove the getters for border style and access the object directly

**Comment by brendandahl** [comment]
Overall looks good, just need to iron the getter/setter stuff.

File: src/core/annotation.js (hunk 1 above)
Comment (timvandermeij) [inline src/core/annotation.js]: Fixed as discussed on IRC.

**Pull request merged**